packages/presentation/lib/screen/home/home_screen.dart

```diff
@@ -24,127 +42,52 @@ class HomeScreen extends StatefulWidget {
   State<HomeScreen> createState() => _HomeScreenState();
 }
 
-class _HomeScreenState extends State<HomeScreen> {
-  @override
-  Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        centerTitle: false,
-        automaticallyImplyLeading: false,
-        backgroundColor: AppColorsDark.primaryColorDark,
-        elevation: 0,
-        title: const Text(
-          'Star Movie',
-          style: TextStyle(
-            fontSize: 24,
-          ),
-        ),
-        actions: <Widget>[
-          GestureDetector(
-            onTap: () {},
-            child: const Icon(
-              Icons.search,
-              size: 35.0,
-            ),
-          ),
-        ],
-      ),
-      body: const HomeTabBar(),
-      bottomNavigationBar: const NavigationBar(),
-    );
-  }
-}
-
-class NavigationBar extends StatelessWidget {
-  const NavigationBar({
-    Key? key,
-  }) : super(key: key);
-
+class _HomeScreenState extends BlocScreenState<HomeScreen, HomeBloc> {
   @override
   Widget build(BuildContext context) {
-    return Container(
-      decoration: const BoxDecoration(
-        border: Border(
-            top: BorderSide(color: AppColorsDark.borderTabBar, width: 1)),
-      ),
-      child: BottomNavigationBar(
-        selectedItemColor: AppColorsDark.selectedBottomNavItem,
-        unselectedItemColor: AppColorsDark.unselectedBottomNavItem,
-        showSelectedLabels: false,
-        showUnselectedLabels: false,
-        items: const [
-          BottomNavigationBarItem(
-            icon: Icon(Icons.person),
-            label: 'Page1',
-          ),
-          BottomNavigationBarItem(
-              icon: Icon(Icons.notifications), label: 'Page2'),
-          BottomNavigationBarItem(
-            icon: Icon(Icons.person),
-            label: 'Page3',
-          ),
-          BottomNavigationBarItem(
-            icon: Icon(Icons.notifications),
-            label: 'Page4',
-          ),
-        ],
-      ),
-    );
-  }
-}
+    return StreamBuilder<BlocData>(
+      stream: bloc.dataStream,
+      builder: (context, snapshot) {
+        final data = snapshot.data;
```

Comment:
pass "HomeData" via generic to avoid using variables like "blocData" with type 'HomeData', because for such purposes we use "tile"

Reply:
you're using two dates, one base, one for the movie, and that's correct, but they must be specified as a generic in "streambuilder", for ease and correctness of use

Reply:
agree with Egor